### PR TITLE
Unit test the main module

### PR DESCRIPTION
**Pull request by WillSewell**
There are a couple of TODOs, but they are more nice to haves rather than things I think will be very helpful right now.

### CODE REVIEW

**Comment by jamescun** [comment]
The output from the tests is quite messy (as expected), might it be worth piping the output to `/dev/null` (`io/ioutil.Discard`); or even capturing it and asserting it outputs what is expected?

File: cmd/buddha_test.go [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+
+// This module is mainly for testing the logic of how the results of Checks
+// are handled
+
+// TODO would also be nice to mock the actual command
+// TODO would also be nice to mock the timeouts
+
+const (
+	DefaultFailures = 2 // Default number of times Checks should be retried
+	DefaultDuration = buddha.Duration(10 ^ 9)
```
Comment (jamescun) [inline cmd/buddha_test.go]: Preferred to write `10 * time.Second`

**Comment by WillSewell** [comment]
Thanks. That's much nicer.

I think checking log output is a bit over the top, but I can pipe to `/dev/null`.

**Comment by WillSewell** [comment]
Should `go get github.com/stretchr/testify/assert` be put in the Makefile?

**Comment by WillSewell** [comment]
I'm actually not sure how to use `log.SetOutput(ioutil.Discard)` with the logger that uses `fmt` to Print which apparently just writes to stdout. It'll be good if you could show me tomorrow.

**Pull request merged**